syncapi/storage/postgres/current_room_state_table.go

```diff
@@ -80,7 +82,7 @@ const selectRoomIDsWithMembershipSQL = "" +
 	"SELECT DISTINCT room_id FROM syncapi_current_room_state WHERE type = 'm.room.member' AND state_key = $1 AND membership = $2"
 
 const selectRoomIDsWithAnyMembershipSQL = "" +
-	"SELECT DISTINCT room_id, membership FROM syncapi_current_room_state WHERE type = 'm.room.member' AND state_key = $1"
+	"SELECT room_id, membership FROM syncapi_current_room_state WHERE type = 'm.room.member' AND state_key = $1"
```

Comment:
Removing `DISTINCT` probably just moves the problem, because now we will spend more time (and context switches) sending duplicates to Dendrite just to filter them out.

Reply:
Shouldn't the `sync_current_room_state` only ever contain one membership event per room?

Reply:
Actually on a second look, it seems `CONSTRAINT syncapi_room_state_unique UNIQUE (room_id, type, state_key)` prevents us from having duplicates already so it should be OK after all.